manager/controllers/app/storage.go

```diff
@@ -18,8 +18,9 @@ import (
 // +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dbuckets,verbs=get;list;watch;update;
 // +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dbuckets/status,verbs=get;update;patch
 
-// FindAvailableBucket finds an available storage asset
-func FindAvailableBucket(c client.Client, log logr.Logger, owner types.NamespacedName, id string, prefixBase string, canShare bool) *app.M4DBucket {
+// FindAvailableBucket finds an available storage asset - in the relevant geo
+// TODO: Implement check for geo.  Needed for ingest
+func FindAvailableBucket(c client.Client, log logr.Logger, owner types.NamespacedName, id string, prefixBase string, canShare bool, geo string) *app.M4DBucket {
```

Comment:
Are we assuming that a bucket can be allocated in any registered cluster?

Reply:
No, that's why I added the geo parameter.  The bucket should be allocated in a cluster in that geo.  Note though that although I added the parameter, the logic was not yet implemented.  I assume that will be done as part of the overall work on issue #104 (provision buckets dynamically)